assets/javascripts/needleeditor.js

```diff
@@ -463,6 +500,15 @@ function saveNeedle(e) {
       addFlash('danger', '<strong>Unable to save needle:</strong><ul><li>' + errors.join('</li><li>') + '</li></ul>');
       return false;
   }
+
+  if ($("#property_workaround").prop("checked") && !$("#input_workaround_desc").val()) {
+    var confirmMessage = "You set a workaround property for this needle but no input description. Are you sure you would not like to input any description for workaround?";
```

Comment:
My suggestion for the text: "You set the workaround property for this needle without a description. Are you sure you want to save without a description?"